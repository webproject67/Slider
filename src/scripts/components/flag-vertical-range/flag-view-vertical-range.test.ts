import SliderModel from '../model/slider-model';
import FlagViewVerticalRange from './flag-view-vertical-range';

const main = document.createElement('div');
const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  range: 'range',
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'vertical'
};

const model = new SliderModel(main, state);
const flag = new FlagViewVerticalRange(model);

test('spyOn mouseDown flag vertical range', () => {
  const somethingSpy = jest.spyOn(flag, 'handleFlagMouseDown');
  let evt: any;
  flag.handleFlagMouseDown(evt);
  expect(somethingSpy).toHaveBeenCalledTimes(1)
});

it('jest snapshots', () => {
  expect(flag.element).toMatchSnapshot()
})