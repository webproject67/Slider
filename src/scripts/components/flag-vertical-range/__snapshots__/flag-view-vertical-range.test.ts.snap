// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest snapshots 1`] = `
<div
  class="slider__flags"
>
  
      
  <span
    class="slider__flag-vertical slider__flag-vertical--min"
    style="top:0%"
  >
    0
  </span>
  
      
  <span
    class="slider__flag-vertical slider__flag-vertical--max"
    data-testid="flagVerticalRange"
    style="top:100%"
  >
    100
  </span>
  
    
</div>
`;
